Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -926,7 +946,7 @@ TextureCacheBase::TCacheEntry* TextureCacheBase::Load(const u32 stage)
 
     // Allocate memory for all levels at once
     size_t total_texture_size = decoded_texture_size;
-    size_t mip_downsample_buffer_size = decoded_texture_size / 4;
+    size_t mip_downsample_buffer_size = decoded_texture_size / 4 * 2;
```

Comment:
Potentially dumb question: what is the multiply by 2 for?

Reply:
I can use less memory there maybe... but the thing is that there are 2 downsample buffers now. I can't reuse the old one because it's the source of the next downsample, so it needs 2 to ping-pong them. 
That "2" was a bit throwaway... the buffers are as big as the first and second mip. So the total size should be `decoded_texture_size *  5 / 16`. I'll change it .

Reply:
Sorry, I have trouble visualizing this stuff, can you explain more where the 5/16th is coming from?

Reply:
So, the Detector takes the first level and downsamples it in buffer1.
Then it downsamples buffer1 into buffer 2.
Then it swaps them and continues this way.

This means that buffer1 needs to be as big as mip level 1, while buffer2 needs to be as big as mip level 2.
This means, that the total memory it needs is `decoded_texture_size / 4 + (decoded_texture_size / 4) / 4`, because those are the sizes of the first 2 mip levels.
With some maths, it becomes `decoded_texture_size * 5 / 16`.